app/shared/github_handler.rb

```diff
@@ -97,5 +101,30 @@ def github_action(client, &block)
         raise ex
       end
     end
+
+    private
+
+    def inject_retry_middleware(client)
+      unless client.middleware.handlers.include?(Faraday::Request::Retry)
+        middleware_dup = client.middleware.dup
+        client.middleware = middleware_dup
+        client.middleware.insert_before(
+          Faraday::Adapter::NetHttp,
+          Faraday::Request::Retry,
+          max: 1000,
```

Comment:
I think max should be lower here. We probably shouldn't go nuts since 1000 tries + exponential backoff means we'll never stop retrying. I think 3 tries is enough, then we'll need to catch the resulting exception.

Reply:
Fair enough! Will change!